explorer/interpreter/action.cpp

```diff
@@ -55,13 +57,21 @@ void RuntimeScope::Initialize(ValueNodeView value_node,
   CARBON_CHECK(!value_node.constant_value().has_value());
   CARBON_CHECK(value->kind() != Value::Kind::LValue);
   allocations_.push_back(heap_->AllocateValue(value));
-  auto [it, success] = locals_.insert(
-      {value_node, heap_->arena().New<LValue>(Address(allocations_.back()))});
+  auto lvalue = heap_->arena().New<LValue>(Address(allocations_.back()));
+  local_values_.push_back(lvalue);
+  auto [it, success] = locals_.insert({value_node, local_values_.size() - 1});
   CARBON_CHECK(success) << "Duplicate definition of " << value_node.base();
 }
 
 void RuntimeScope::Merge(RuntimeScope other) {
   CARBON_CHECK(heap_ == other.heap_);
+  for (auto& element : other.locals_) {
+    if (local_values_.size() > 0)
+      other.locals_[element.first] += local_values_.size();
```

Comment:
```suggestion
    if (local_values_.size() > 0) {
      other.locals_[element.first] += local_values_.size();
    }
```

Minor style detail:
"Always use braces for conditional, switch, and loop statements, even when the body is a single statement."

https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/cpp_style_guide.md#syntax-and-formatting